R/load.project.R

```diff
@@ -211,6 +211,15 @@ load.project <- function(override.config = NULL)
                      filename,
                      variable.name))
 
+        ## Most times, variable.name will correspond to one loaded object
+        ## but in the case of xls(x) readers, they always return
+        ## variable.name.Sheet1 ... variable.name.SheetN
+        if (grepl('xls', extension)) {
+          variable.name <- grep(paste(variable.name, 'Sheet', sep ='.'),
+                                ls(.TargetEnv),
+                                value = TRUE)
```

Comment:
This looks brittle to me. Can the XLS reader (and also other readers that create multiple variables) return a list of variable names actually created?

I spent some time trying to create the LoadMyData package from ideas in ProjectTemplate. It's certainly possible to overengineer a solution for the data loading problem. I've come to the conclusion that the concept of automatically loading data only works for "standard" formats such as R data files. For about anything else, there's usually more than one way to read the data, and for some inputs one way works better than the other. Plus, there are options (such as extracting a specific region from an Excel sheet).

I'm not sure what to do in the context of ProjectTemplate. For each new data file added by the user, we could create a small `.R` file that contains code that loads the file, just like it's loaded now. The convention would be that for each data file there must be a corresponding `.R` file. @KentonWhite: What do you think?

Back to this pull request: If all data are read by R scripts, then the output of these R scripts defines the new variable names. These scripts can be sourced in a new environment, and whatever ends in this environment are the new variables.